### DISCUSSION TITLE
[feature_request]  Add date selector arguments to GraphQL repositoriesContributedTo object

### DISCUSSION THREAD

**Discussion by rickstaa** [community/community#39589] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

I'm trying to get the total number of repositories a user has contributed to since GitHub launched in 2008. My first impression was that I could get this number from GraphQLs `repositoriesContributedTo.totalCount` object. As explained in https://github.com/orgs/community/discussions/24350 and [the  GraphQL documentation](https://docs.github.com/en/graphql/reference/objects#user), however, this object seems to give the results for the last year. Since this object doesn't have a `from` and `to` argument, I could not find an easy way to loop through all years to get the total `repositoriesContributedTo`. For my use case, it would be fantastic if you could change the period of the `repositoriesContributedTo.totalCount` object from GitHub's onset to the current time. Alternatively, adding the date selector arguments (i.e. `from` and `to`) to the `repositoriesContributedTo.totalCount` object would significantly improve the user experience.

#### Current behaviour

The only way I currently found for retrieving the total number of PUBLIC repositories a user has contributed to is looping through all years in the `contributionsCollection.totalRepositoriesWithContributedCommits` object via the `from` and `to` arguments. 

```graphql
query userInfo($login: String!) {
  user(login: $login) {
    name
    repositoriesContributedTo(
      contributionTypes: [COMMIT, ISSUE, PULL_REQUEST, REPOSITORY]
    ) {
      totalCount
    }
    contributionsCollection(from:"2020-11-20T15:45:00.000Z", to:"2021-11-20T15:45:00.000Z") {
      totalRepositoriesWithContributedCommits
    }
  }
}
```
```graphql
{"login": "rickstaa"}
```

This, however, also includes the repositories a user owns, so I still need to filter these repositories out, which costs additional GraphQL points and requires unnecessary code.

#### New behaviour

It would be great if you could provide the `repositoriesContributedTo` directly like in the `followers` object:

```graphql
query userInfo($login: String!) {
  user(login: $login) {
    name
    repositoriesContributedTo(
      contributionTypes: [COMMIT, ISSUE, PULL_REQUEST, REPOSITORY]
    ) {
      totalCount
    }
  }
}
```
```graphql
{"login": "rickstaa"}
```

Or add the date selector arguments to the `repositoriesContributedTo` object:

```graphql
query userInfo($login: String!) {
  user(login: $login) {
    name
    repositoriesContributedTo(
      from:"2020-11-20T15:45:00.000Z", to:"2021-11-20T15:45:00.000Z",
      contributionTypes: [COMMIT, ISSUE, PULL_REQUEST, REPOSITORY]
    ) {
      totalCount
    }
  }
}
```
```graphql
{"login": "rickstaa"}
```

**Comment by gxanshu** [comment]
```gql
query userInfo($login: String!) {
  user(login: $login) {
    name
    repositoriesContributedTo(
      from:"2020-11-20T15:45:00.000Z", to:"2021-11-20T15:45:00.000Z",
      contributionTypes: [COMMIT, ISSUE, PULL_REQUEST, REPOSITORY]
    ) {
      totalCount
    }
  }
}
```

running this code gives a error `Field 'repositoriesContributedTo' doesn't accept argument 'from`

**Reply by dargmuesli** [reply]
This must be because it's a feature *request* 💡🌈

**Reply by gxanshu** [reply]
my bad, sorry about that. hope this feature will came soon

**Reply by Ifiht** [reply]
bump, please make it so

**Reply by udlose** [reply]
would love to see this implemented!

**Comment by dargmuesli** [comment]
:mega: I've created a Python script to extract JSON data from the profile pins HTML in the meantime: https://github.com/dargmuesli/contributed-to 😅
